snapcraft/commands/remote.py

```diff
@@ -211,33 +186,96 @@ def _run_new_or_fallback_remote_build(self, base: str) -> None:
         emit.debug("Running fallback remote-build.")
         run_legacy()
 
-    @overrides
-    def run(self, parsed_args: argparse.Namespace) -> None:
-        """Run the remote-build command.
+    def _get_project_name(self) -> str:
+        """Get the project name from the project's snapcraft.yaml.
 
-        :param parsed_args: Snapcraft's argument namespace.
+        :returns: The project name.
 
-        :raises AcceptPublicUploadError: If the user does not agree to upload data.
-        :raises SnapcraftError: If the project cannot be loaded and parsed.
+        :raises SnapcraftError: If the snapcraft.yaml does not contain a 'name' keyword.
         """
-        if os.getenv("SUDO_USER") and os.geteuid() == 0:
-            emit.message(
-                "Running with 'sudo' may cause permission errors and is discouraged."
+        with open(self._snapcraft_yaml, encoding="utf-8") as file:
+            data = yaml_utils.safe_load(file)
+
+        project_name = data.get("name")
+
+        if project_name:
+            emit.debug(
+                f"Using project name {project_name!r} from "
+                f"{str(self._snapcraft_yaml)!r}."
             )
+            return project_name
 
-        emit.message(
-            "snapcraft remote-build is experimental and is subject to change "
-            "- use with caution."
+        raise SnapcraftError(
+            f"Could not get project name from {str(self._snapcraft_yaml)!r}."
         )
 
-        if parsed_args.build_on:
-            emit.message("Use --build-for instead of --build-on")
-            parsed_args.build_for = parsed_args.build_on
+    def _run_new_remote_build(self) -> None:
+        """Run new remote-build code."""
+        # the build-id will be passed to the new remote-build code as part of #4323
+        if self._parsed_args.build_id:
+            build_id = self._parsed_args.build_id
+            emit.debug(f"Using build ID {build_id!r} passed as a parameter.")
+        else:
+            build_id = get_build_id(
+                app_name="snapcraft",
+                project_name=self._get_project_name(),
+                project_path=Path(),
+            )
+            emit.debug(f"Using computed build ID {build_id!r}.")
 
-        if not parsed_args.launchpad_accept_public_upload and not confirm_with_user(
-            _CONFIRMATION_PROMPT
-        ):
-            raise AcceptPublicUploadError()
+        # TODO: use new remote-build code (#4323)
```

Comment:
If `build_id` is used to reuse an existing git repo, does it make sense to include files under `.git` when calculating the id? We could save some time and cpu skipping it.

Reply:
Yes, we could ignore `.git` to improve performance. I am going to leave it as-is for now.